index.html

```diff
@@ -3,449 +3,295 @@
+        Processing an image resource from JSON
+      </h2>
       <p>
-        The steps for <dfn>fetching an image</dfn> are given by the
-        following algorithm. The algorithm takes a <a>processed image
-        resource</a> <var>image</var>, and an optional <a>Request</a>
-        <var>request</var>. It returns a <a>Response</a>.
+        The steps to <dfn data-export="">process an image resource from
+        JSON</dfn> are given by the following algorithm. The algorithm takes a
+        JSON Object |input:Object| and a [=URL=] |base URL|. It returns an
+        [=image resource=].
       </p>
       <ol>
-        <li>If <var>request</var> is <code>undefined</code>:
-          <ol>
-            <li>Set <var>request</var> to a new <a>Request</a>.
-            </li>
-            <li>Set the following properties of <var>request</var>:
-              <ol>
-                <li>[=request/URL=] is set to <var>image</var>["src"].
-                </li>
-                <li>[=request/Destination=] is set to "image".
-                </li>
-                <li>[=request/Mode=] is set to "no-cors".
-                </li>
-              </ol>
-            </li>
-          </ol>
+        <li>If |input| is missing "src" property, return failure.
         </li>
-        <li>Perform a <a>fetch</a> using <var>request</var> and return the
-        <var>response</var>.
+        <li>Let |image| be a new [=image resource=].
         </li>
-      </ol>
-    </section>
-    <section>
-      <h3>
-        Terms defined by reference
-      </h3>
-      <ul class="index" data-sort="">
-        <li>[[URL]] defines the following terms:
-          <ul>
-            <li>
-              <dfn data-cite="URL#concept-url">URL</dfn>
-            </li>
-            <li>
-              <dfn data-cite="URL#concept-url-parser" data-lt=
-              "parse|parsing|URL parsing">URL parser</dfn>
-            </li>
-          </ul>
+        <li>Set |image|'s [=image resource/src=] to the result of [=URL
+        parser|URL parsing=] |object|'s [=image resource/src=] property and
+        |base URL|. If parsing the URL fails, return failure.
         </li>
-        <li>[[FETCH]] defines the following terms:
-          <ul data-sort="">
-            <li>
-              <dfn data-cite="FETCH#concept-fetch">Fetch</dfn>
-            </li>
-            <li>
-              <dfn data-cite="FETCH#concept-request">Request</dfn>
-            </li>
-          </ul>
-        </li>
-        <li>[[APPMANIFEST]] defines the following terms:
-          <ul>
-            <li>
-              <a data-cite="APPMANIFEST#manifest"><dfn>manifest</dfn></a>
+        <li>If |input| has a "sizes" property and its type is a [=string=]
+        of length greater than zero:
+          <ol>
+            <li>Set |image|'s [=image resource/sizes=] to the result of parsing
+            |input|'s "sizes" property value as if it was a [^link/sizes^]
+            attribute. If parsing fails, return failure.
```

Comment:
There's a lot of repetition of the previous algorithm here. Any way to avoid it?